File: src/modules/stored_file/services/stored_file_delete_service.py
from src.infra.file_system.implementations.aws_file_system import AwsFileSystem
from src.modules.stored_file.models import StoredFile
from src.shared.exceptions.not_found_exception import NotFoundException


class StoredFileDeleteService:
    @classmethod
    def execute(self, uuid: str):
        stored_file = StoredFile.objects.filter(uuid=uuid).first()

        if stored_file is None:
            raise NotFoundException("Arquivo não encontrado!")

        AwsFileSystem.delete(f"{stored_file.relative_path}/{stored_file.stored_name}")

        stored_file.delete()

File: src/modules/stored-file/services/stored-file-delete.service.ts
import { prisma } from '../../../infra/database/prisma.service';
import { awsFileSystem } from '../../../infra/file-system/implementations/aws.file-system';
import { NotFoundException } from '../../../shared/exceptions';

class StoredFileDeleteService {
  async execute(uuid: string): Promise<void> {
    const storedFile = await prisma.storedFile.findUnique({
      where: { uuid },
    });

    if (!storedFile) throw new NotFoundException('Arquivo não encontrado!');

    await awsFileSystem.delete(
      `${storedFile.relativePath}/${storedFile.storedName}`,
    );

    await prisma.storedFile.delete({ where: { uuid } });
  }
}
export const storedFileDeleteService = new StoredFileDeleteService();
